### PR TITLE
some changes

### PR DESCRIPTION
**Pull request by DavidBJaffe**
- major reorg into workspace, thus speeding up compilation (more to come)
- add a Windows page
- add a "how to set your path" page
- add explanation of PRE
- properly expand ~ in paths (not sure when this was working before)
- add a "build" script to enable one-stop building
- add test for license
- add stat n_twothreesie = number of cells having 2 or 3 chains
- nicify an internal error message

### CODE REVIEW

**Comment by meryllewis** [comment]
jenkins retest this please

**Comment by meryllewis** [comment]
jenkins retest this please

**Comment by DavidBJaffe** [comment]
jenkins retest this please

File: enclone/src/proc_args2.rs [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+            eprintln!("\nInput dataset path {} is duplicated.\n", dp[i]);
+            std::process::exit(1);
+        }
+        i = j;
+    }
+    if !ctl.silent {
+        println!("");
+    }
+
+    // Get sample descriptions.  Flaky and particularly flaky when lena args are paths,
```
Comment (wyattmcdonnell) [inline enclone/src/proc_args2.rs:149]:
```suggestion
    // Get sample descriptions.  Flaky and particularly flaky when internal sample args are paths,
```

**Comment by meryllewis** [comment]
jenkins retest this please

File: enclone_main/src/main_enclone.rs [hunk 2]
```diff
@@ -35,6 +44,206 @@ use std::{
...
+    }
+
+    // Process args (and set defaults for them).
+
+    proc_args(&mut ctl, &args);
+
+    // Dump lenas.
+
+    for i in 1..args.len() {
+        if is_simple_arg(&args[i], "DUMP_LENAS") {
```
Comment (wyattmcdonnell) [inline enclone_main/src/main_enclone.rs:233]: Flagging for renaming

**Comment by meryllewis** [comment]
jenkins retest this please

**Comment by meryllewis** [comment]
jenkins retest this please

**Comment by meryllewis** [comment]
jenkins retest this please

File: enclone_main/src/main_enclone.rs (hunk 2 above)
Comment (DavidBJaffe) [reply enclone_main/src/main_enclone.rs]: Fixing separately!

**Pull request merged**